### ISSUE TITLE
Skip "new element" dialog if there is only one (allowed) Node Type

### ISSUE THREAD

**Issue by bwaidelich** [neos/neos-development-collection#2174]
As suggested here: https://neos-project.slack.com/archives/C0U0KEGDQ/p1513932060000034

**Comment by aertmann** [comment]
That's already the case in the old UI as far as I remember, could be wrong though. At least the idea has come up several times.

**Comment by bwaidelich** [comment]
> That's already the case in the old UI as far as I remember

I don't think that that was the case tbh.

> the idea has come up several times.

Yes, but I couldn't find a ticket for it. Shouldn't be too hard to implement this (hopefully)

**Comment by aertmann** [comment]
True, can confirm it's not the case in the old UI.

As for a ticket, lots of tickets have gotten lost in the Redmine -> Jira -> Github transitions.

Think it's super easy to implement.

**Comment by dimaip** [comment]
The problem here is that in the nodetype selection dialog we now also choose the insertion mode, which can't be skipped.

**Comment by bwaidelich** [comment]
> The problem here is that in the nodetype selection dialog we now also choose the insertion mode, which can't be skipped.

Ah right, I forgot about that one.. Mh, too bad. But maybe we can find a solution for that?

**Comment by dimaip** [comment]
A possible solution could be that if for a given position there's only one insertion mode available (that will always be `into`) AND only one nodetype availabe, THEN skip the dialog.

**Comment by bwaidelich** [comment]
>  if for a given position there's only one insertion mode available [...] THEN skip the dialog.

yes, that makes sense. Although that's probably not too often the case.
Do you guys remember why we moved the insert mode to the dialog in the first place?

**Comment by dimaip** [comment]
Because it was very hard and confusing for editors to use the dropdown in the tree.
I wasn't the one who moved it, but I improved its styling within the modal.

**Comment by aertmann** [comment]
I could easily be the case if you have an empty content collection inside a page or an element that only allows one type of element, here it would make sense and not that uncommon use case.

**Comment by bwaidelich** [comment]
FYI: I realized that 99% of our cases "inside" would be the only option anyways (due to extensive use of constraints) so this would be a really great improvement already

**Comment by jonnitto** [comment]
Need this some work to do or it is ready for merge?
